File: src/Advertiser.java
import java.util.ArrayList;

public class Advertiser extends BaseAdvertising {

    private String name;
    private static ArrayList<Advertiser> allAdvertisers = new ArrayList<>();

    public Advertiser(int id, String name) {
        super(id);
        this.name = name;
        allAdvertisers.add(this);
    }

    public static int getTotalClicks() {
        int sum = 0;
        for (Advertiser advertiser : allAdvertisers) {
            sum += advertiser.getClicks();
        }
        return sum;
    }

    public static String help() {
        return "I have id, name, clicks and views fields";
    }
    public String describeMe() {
        return "I am advertiser class :)";
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }


}

File: python/Advertiser.py
from main import BaseAdvertising


class Advertiser(BaseAdvertising):
    __total_clicks = 0

    def __init__(self, id, name):
        super().__init__(id)
        self.__name = name

    @staticmethod
    def help():
        return "I have id, name, clicks and views fields"

    @staticmethod
    def describe_me():
        return "I am advertiser class :)"

    def get_name(self):
        return self.__name

    def set_name(self, name):
        self.__name = name

    @staticmethod
    def get_total_clicks():
        return Advertiser.__total_clicks

    @staticmethod
    def inc_total_clicks():
        Advertiser.__total_clicks += 1
